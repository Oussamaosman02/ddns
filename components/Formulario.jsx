import React, { useRef } from 'react'
import inversa from './fn/inversa'

function Formulario ({ data }) {
  const ndominio = useRef()
  const nservpri = useRef()
  const nservsec = useRef()
  const nservotro = useRef()
  const nmascara = useRef()
  const nmaestre = useRef()
  const nftp = useRef()

  function handleSubmit (e) {
    e.preventDefault()

    const nombreDeDominio = ndominio.current.value
    const ipNormal = nservpri.current.value
    const ipSecundario = nservsec.current.value
    const ipMaestre = nservotro.current.value
    const mascara = nmascara.current.value
    const maestreNombre = nmaestre.current.value
    const ipFtp = nftp.current.value

    if (nombreDeDominio && ipNormal && ipSecundario && ipMaestre && mascara) {
      const IPZonaInversa = inversa(mascara, ipNormal, true)
      const ipInversa = inversa(mascara, ipNormal)
      const secundarioInversa = inversa(mascara, ipSecundario)
      const maestreInversa = inversa(mascara, ipMaestre)
      const ftpInversa = inversa(mascara, ipFtp)

      const result = {
        nombreDeDominio,
        ipNormal,
        ipSecundario,
        ipMaestre,
        mascara,
        ipInversa,
        IPZonaInversa,
        secundarioInversa,
        maestreInversa,
        maestreNombre,
        ftpInversa,
        ipFtp
      }
      data(result)
    } else {
      // eslint-disable-next-line no-undef
      alert('introduce todos los parámetros')
    }
  }

  return (
    <form onSubmit={handleSubmit} aria-label='form'>
      <h2>Introduce los datos</h2>
      <input placeholder='nombre de dominio' ref={ndominio} />
      <input placeholder='ip servidor primario' ref={nservpri} />
      <input placeholder='ip servidor secundario' ref={nservsec} />
      <input placeholder='nombre otro host' ref={nmaestre} />
      <input placeholder='ip otro host' ref={nservotro} />
      <input placeholder='ip ftp' ref={nftp} />
      <input placeholder='mascara 8-16-24' ref={nmascara} />
      <button>Generar documentación</button>
    </form>
  )
}

export default Formulario
