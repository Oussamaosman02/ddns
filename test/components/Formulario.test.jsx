import { screen, render, fireEvent } from '@testing-library/react'
import Formulario from '../../components/Formulario'
import { test, describe, expect, jest } from '@jest/globals'

describe('Test de <Formulario/>', () => {
  const handle = jest.fn()
  const data = {
    nombreDeDominio: 'dominio.asir2',
    ipNormal: '10.8.23.5',
    ipInversa: '5.23',
    IPZonaInversa: '8.10',
    ipSecundario: '10.8.23.7',
    secundarioInversa: '7.23',
    ipMaestre: '10.8.23.9',
    maestreInversa: '9.23',
    maestreNombre: 'nombre',
    ipFtp: '10.8.23.15',
    ftpInversa: '15.23',
    mascara: '16'
  }
  const { nombreDeDominio, ipNormal, ipSecundario, maestreNombre, ipMaestre, ipFtp, mascara } = data

  test('debe coincidir con el snapshot', () => {
    const { container } = render(<Formulario data={handle} />)
    expect(container).toMatchSnapshot()
  })
  test('debe activar el handleSubmit', () => {
    render(<Formulario data={handle} />)
    const form = screen.getByRole('form')
    function disparar (placeholder, value) {
      const input = screen.getByPlaceholderText(placeholder)
      fireEvent.input(input, { target: { value } })
    }
    disparar('nombre de dominio', nombreDeDominio)
    disparar('ip servidor primario', ipNormal)
    disparar('ip servidor secundario', ipSecundario)
    disparar('nombre otro host', maestreNombre)
    disparar('ip otro host', ipMaestre)
    disparar('ip ftp', ipFtp)
    disparar('mascara 8-16-24', mascara)

    fireEvent.submit(form)
    expect(handle).toHaveBeenCalled()
  })
  test('debe devolver los valores correctos', () => {
    render(<Formulario data={handle} />)
    const form = screen.getByRole('form')
    function disparar (placeholder, value) {
      const input = screen.getByPlaceholderText(placeholder)
      fireEvent.input(input, { target: { value } })
    }
    disparar('nombre de dominio', nombreDeDominio)
    disparar('ip servidor primario', ipNormal)
    disparar('ip servidor secundario', ipSecundario)
    disparar('nombre otro host', maestreNombre)
    disparar('ip otro host', ipMaestre)
    disparar('ip ftp', ipFtp)
    disparar('mascara 8-16-24', mascara)

    fireEvent.submit(form)
    expect(handle.mock.lastCall[0]).toEqual(data)
  })
})
