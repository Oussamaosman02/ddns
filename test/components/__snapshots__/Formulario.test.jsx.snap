// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de <Formulario/> debe coincidir con el snapshot 1`] = `
<div>
  <form
    aria-label="form"
  >
    <h2>
      Introduce los datos
    </h2>
    <input
      placeholder="nombre de dominio"
    />
    <input
      placeholder="ip servidor primario"
    />
    <input
      placeholder="ip servidor secundario"
    />
    <input
      placeholder="nombre otro host"
    />
    <input
      placeholder="ip otro host"
    />
    <input
      placeholder="ip ftp"
    />
    <input
      placeholder="mascara 8-16-24"
    />
    <button>
      Generar documentaci√≥n
    </button>
  </form>
</div>
`;
